easybuild/easyblocks/generic/systemcompiler.py

```diff
@@ -168,3 +258,35 @@ def make_module_extend_modpath(self):
         # Reset to actual compiler version (e.g., "4.8.2")
         self.cfg['version'] = self.compiler_version
         return res
+
+    def make_module_extra(self, *args, **kwargs):
+        """Add any additional module text."""
+        if self.cfg['generate_standalone_module']:
+            if self.cfg['name'] in ['GCC','GCCcore']:
+                extras = EB_GCC.make_module_extra(self, *args, **kwargs)
+            elif self.cfg['name'] in ['icc']:
+                extras = EB_icc.make_module_extra(self, *args, **kwargs)
+            elif self.cfg['name'] in ['ifort']:
+                extras = EB_ifort.make_module_extra(self, *args, **kwargs)
+            else:
+                raise EasyBuildError("I don't know how to generate extra module text for %s", self.cfg['name'])
+        else:
+            extras = super(SystemCompiler, self).make_module_extra(*args, **kwargs)
```

Comment:
@boegel For a system `icc` this seems to kick an error:
```
File "/tmp/eb-JeUdnR/included-easyblocks/easybuild/easyblocks/generic/systemcompiler.py", line 290, in sanity_check_step
    fake_mod_data = self.load_fake_module(purge=True)
  File "/usr/local/software/jureca/Stages/2017a/software/EasyBuild/3.3.1/lib/python2.7/site-packages/easybuild_framework-3.3.1-py2.7.egg/easybuild/framework/easyblock.py", line 1251, in load_fake_module
    fake_mod_path = self.make_module_step(fake=True)
  File "/tmp/eb-JeUdnR/included-easyblocks/easybuild/easyblocks/generic/systemcompiler.py", line 240, in make_module_step
    res = super(SystemCompiler, self).make_module_step(fake=fake)
  File "/usr/local/software/jureca/Stages/2017a/software/EasyBuild/3.3.1/lib/python2.7/site-packages/easybuild_framework-3.3.1-py2.7.egg/easybuild/framework/easyblock.py", line 2195, in make_module_step
    txt += self.make_module_extra()
  File "/tmp/eb-JeUdnR/included-easyblocks/easybuild/easyblocks/generic/systemcompiler.py", line 274, in make_module_extra
    extras = super(SystemCompiler, self).make_module_extra(*args, **kwargs)
  File "/usr/local/software/jureca/Stages/2017a/software/EasyBuild/3.3.1/lib/python2.7/site-packages/easybuild_easyblocks-3.3.1-py2.7.egg/easybuild/easyblocks/generic/bundle.py", line 165, in make_module_extra
    return super(Bundle, self).make_module_extra(altroot=self.altroot, altversion=self.altversion)
TypeError: make_module_extra() got an unexpected keyword argument 'altroot'
```

Reply:
This error only occurs when `generate_standalone_module = False`, otherwise it works

Reply:
Should I be doing nothing if `generate_standalone_module = False` to avoid this problem?

Reply:
You're not using the customised `bundle.py` easyblock from this PR which does `EasyBlock.make_module_extra(...)` rather than `super(Bundle, self).make_module_extra(...)`

Reply:
I ran into this same problem when testing the `SystemMPI` easyblock, and it was fixed when also using the updated `bundle.py`.

Reply:
Yes, confirmed, works with the `bundle.py` in the PR